src/Mod/Path/libarea/Adaptive.cpp

```diff
@@ -146,6 +146,38 @@ inline DoublePoint GetPathDirectionV(const Path &pth, size_t pointIndex)
 	const IntPoint &p2 = pth.at(pointIndex);
 	return DirectionV(p1, p2);
 }
+	
+bool doFilter(Path &ppg) {
+        size_t i;
+        size_t b;
+        size_t start;
+        size_t end;
+        for(i=0;i<ppg.size();i++) {
+            if (i==ppg.size()-1) {
+                start=0;
+                end=ppg.size()-2;
+            } else {
+                start=i+1;
+                end=ppg.size();
+            }
+            for(b=start;b<end;b++) {
+                if (abs(ppg[i].X-ppg[b].X)<2 && abs(ppg[i].Y-ppg[b].Y)<2) {
```

Comment:
The 2 seems arbitrary here: through what mechanism is "2" defined as "close"?

Reply:
if the nearby coordinate is the same or only differs by 1 unit just remove it, the vertexes are upscaled to a relatively high number (clipper uses integer only) so they can just be removed. I'm using Path Adaptive a lot and I was also able to trigger this problem easily as well.